src/precice/impl/WatchPoint.cpp

```diff
@@ -61,13 +59,15 @@ void WatchPoint::initialize()
   _weights.clear();
   _vertices.clear();
 
+  mesh::Vertex watchVertex(_point, 0);
```

Comment:
Having to create a dummy point is a strong indicator for the interface of `query::Index` being incomplete.
Looking at it, there is really no reason why the `Index::getClosestX` functions require a full Vertex as argument (aside from convenience).

My suggestion for now:
* We add overloads that take `const Eigen::VectorXd&` instead of `const mesh::Vertex&`.
* We can then implement the Vertex version then using the Eigen version.